websockets/bridge.go

```diff
@@ -6,62 +6,14 @@ import (
-	"github.com/buildwithgrove/path/observation"
 )
 
-// TODO_TECHDEBT(@adshmh): Refactor the bridge to purely contain any required specialized websocket logic.
-// - Most of the logic here needs to move to a new, specialized gateway.WebsocketRequestContext struct.
-//
-// TODO_TECHDEBT(@adshmh): make this a package private struct.
-// No other package should need to know the details of the underlying protocol (Websocket in this case).
-//
-// WebSocketMessage represents a websocket message that can be one of the following:
-// - Client request
-// - Endpoint response
-// - Subscription push event (e.g. eth_subscribe)
-type WebSocketMessage struct {
-	// Data is the message payload
-	Data []byte
-
-	// MessageType is an int returned by the gorilla/websocket package
-	MessageType int
-}
-
-// TODO_TECHDEBT(@adshmh): Rename/remove/split this into multiple interfaces.
-// Each interface's name should clearly indicate the responsibilities.
-//
 // WebSocketMessageHandler handles websocket messages.
 // It can, for example, transform the message data before forwarding it.
 type WebSocketMessageHandler interface {
 	// HandleMessage processes a message and returns the data to forward.
 	// If an error is returned, the connection will be closed.
-	HandleMessage(msg WebSocketMessage) ([]byte, error)
-}
-
-// TODO_TECHDEBT(@adshmh): Drop this interface once the following are complete:
-// - The gateway package is the sole publisher of observations.
-// - Each message from the client/endpoint should result in a new observation (some fields will have the same values, e.g. the endpoint address)
-// - Each layer (e.g. protocol, QoS) should build and supply its own specialized observations for each message.
-//
-// ObservationPublisher handles publishing observations after processing endpoint messages.
-type ObservationPublisher interface {
-	// SetObservationContext sets the gateway observations and data reporter.
-	// Set once per Bridge initialization.
-	SetObservationContext(*observation.GatewayObservations, gateway.RequestResponseReporter)
-	// InitializeMessageObservations initializes the observations for the current message.
-	// Called once per message.
-	InitializeMessageObservations() *observation.RequestResponseObservations
-	// UpdateMessageObservations updates the observations for the current message.
-	// Called once per message if the message handler does not return an error.
-	UpdateMessageObservationsFromSuccess(*observation.RequestResponseObservations)
-	// UpdateMessageObservationsFromError updates the observations for the current message.
-	// Called once per message if the message handler returns an error.
-	UpdateMessageObservationsFromError(*observation.RequestResponseObservations, error)
-	// PublishObservations publishes protocol-specific observations.
-	// Called once per message.
-	PublishMessageObservations(*observation.RequestResponseObservations)
+	HandleMessage(msgData []byte) ([]byte, error)
```

Comment:
Please revise to clarify what this method does.

Reply:
have clarified in name and with comments.